config/secure-headers.php

```diff
@@ -375,43 +390,165 @@
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/prefetch-src
+		'prefetch-src' => [
+		],
+
+		// https://w3c.github.io/webappsec-trusted-types/dist/spec/#integration-with-content-security-policy
+		'require-trusted-types-for' => [
+			'script' => false,
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/sandbox
+		'sandbox' => [
+			'enable' => false,
+
+			'allow-downloads-without-user-activation' => false,
+
+			'allow-forms' => false,
+
+			'allow-modals' => false,
+
+			'allow-orientation-lock' => false,
+
+			'allow-pointer-lock' => false,
+
+			'allow-popups' => false,
+
+			'allow-popups-to-escape-sandbox' => false,
+
+			'allow-presentation' => false,
+
+			'allow-same-origin' => false,
+
+			'allow-scripts' => false,
+
+			'allow-storage-access-by-user-activation' => false,
+
+			'allow-top-navigation' => false,
+
+			'allow-top-navigation-by-user-activation' => false,
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src
+		'script-src' => [
+			'none' => false,
+
+			'self' => true,
+
+			// https://www.chromestatus.com/feature/5792234276388864
+			'report-sample' => true,
+
 			'allow' => [
-				'http://lycheeorg.github.io/update.json',
-				'blob:',
+				'https://www.dropbox.com/static/api/1/dropins.js',
+				// 'url',
 			],
-			'self' => true,
+
+			'schemes' => [
+				// 'data:',
+				// 'https:',
+			],
+
+			/* followings are only work for `script` and `style` related directives */
+
+			'unsafe-inline' => false,
+
+			'unsafe-eval' => false,
+
+			// https://www.w3.org/TR/CSP3/#unsafe-hashes-usage
+			'unsafe-hashes' => false,
+
+			// Enable `strict-dynamic` will *ignore* `self`, `unsafe-inline`,
+			// `allow` and `schemes`. You can find more information from:
+			// https://www.w3.org/TR/CSP3/#strict-dynamic-usage
+			'strict-dynamic' => false,
+
+			'hashes' => [
+				'sha256' => [
+					// 'sha256-hash-value-with-base64-encode',
+				],
+
+				'sha384' => [
+					// 'sha384-hash-value-with-base64-encode',
+				],
+
+				'sha512' => [
+					// 'sha512-hash-value-with-base64-encode',
+				],
+			],
+
+			'nonces' => [
+				// 'base64-encoded',
+			],
+
+			'unsafe-hashed-attributes' => false,
+
+			'add-generated-nonce' => false,
 		],
-		'form-action' => [
-			'self' => true,
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src-attr
+		'script-src-attr' => [
 		],
-		'frame-ancestors' => [],
-		'frame-src' => [],
-		'manifest-src' => [],
-		'media-src' => [
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src-elem
+		'script-src-elem' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src
+		'style-src' => [
 			'self' => true,
-			'allow' => [
-				'blob:',
+			'unsafe-inline' => true, // We sadly need this one
```

Comment:
We should make the comment more specific. Nobody will know _why_ we need it in two month from now. I would also remove the appraisement ("sadly"). Better something like

```php
// TODO: Set this to `false` after the inline style in <reference here> has been moved to a style file
```

Reply:
Agreed re more specific, but we can't be *that* specific.

Maybe...
```suggestion
			'unsafe-inline' => true, // We sadly need this one due to styles applied by javascript
```

Reply:
I assumed that only inline styles are problematic as we have in

https://github.com/LycheeOrg/Lychee-front/blob/692dbacabb739756a2027c8f95f7e9b59e93dc16/scripts/main/settings.js#L472-L473

JavaScript should only add or remove style classes as in

https://github.com/LycheeOrg/Lychee-front/blob/692dbacabb739756a2027c8f95f7e9b59e93dc16/scripts/main/settings.js#L220

I thought that this would be fine and does not count as "inline style". If this is indeed the case, there are really only a handful of inline styles which need to be removed. (Actually, I am only aware of the single one above )

Reply:
> JavaScript should only add or remove style classes as in

I was basing this on @ildyria's  response on gitter, but there do appear to be some direct references to the `style` property and a few inline classes if you `git grep style` (when at a pc).

Reply:
OK, I did not want to imply that the reference which I have been aware of was the only one. I only wanted to point out that JavaScript wasn't a general problem as long as JS stuck to only modify style classes.

So maybe something like this

```
// We need this one due to direct styles (not just style classes) applied by javascript
```

Reply:
Fixed this one myself in https://github.com/LycheeOrg/Lychee/pull/1528/commits/757a77e7b077955aef2afbd8e4829ce1b38aee20.